File: src/main/java/sage/transfer/BlogView.java
package sage.transfer;

import java.util.Date;
import java.util.List;

import sage.entity.Blog;
import sage.entity.User;
import sage.util.Colls;

public class BlogView {

  private Long id;
  private UserLabel author;
  private String title;
  private String content;
  private Date createdTime;
  private Date modifiedTime;
  private List<TagLabel> tags;
  
  BlogView() {}

  public BlogView(Blog blog) {
    id = blog.getId();
    author = new UserLabel(blog.getAuthor());

    title = blog.getTitle();
    content = blog.getContent();
    createdTime = blog.getCreatedTime();
    modifiedTime = blog.getModifiedTime();

    tags = Colls.map(blog.getTags(), TagLabel::new);
  }

  public Long getId() {
    return id;
  }

  public UserLabel getAuthor() {
    return author;
  }

  public String getTitle() {
    return title;
  }

  public String getContent() {
    return content;
  }

  public Date getCreatedTime() {
    return createdTime;
  }

  public Date getModifiedTime() {
    return modifiedTime;
  }

  public List<TagLabel> getTags() {
    return tags;
  }
}

File: src/main/kotlin/sage/transfer/BlogView.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import sage.entity.Blog
import java.sql.Timestamp

@KotlinNoArg
class BlogView {
  var id: Long = 0
  var author: UserLabel? = null
  var title: String = ""
  var contentType: String = ""
  var inputContent: String = ""
  var content: String = ""
  var whenCreated: Timestamp? = null
  var whenEdited: Timestamp? = null
  var tags: List<TagLabel> = arrayListOf()
  var tweetId: Long = 0

  var likes: Int = 0
  var views: Int = 0

  constructor()

  constructor(blog: Blog) : this(blog, false)

  constructor(blog: Blog, showInputContent: Boolean) {
    id = blog.id
    author = blog.author.toUserLabel()

    title = blog.title
    contentType = Blog.contentTypeString(blog.contentType)
    if (showInputContent) inputContent = blog.inputContent
    content = blog.content
    whenCreated = blog.whenCreated
    whenEdited = actualWhenEdited(blog.whenCreated, blog.whenEdited)

    tags = blog.tags.map { it.toTagLabel() }
    tweetId = blog.tweetId

    blog.stat()?.let {
      likes = it.likes
      views = it.views
    }
  }
}
